### PR TITLE
Implements subscriptions and community alerts page.

### PR DESCRIPTION
**Pull request by burtonator**
## Link to Issue
Closes: #6627
Closes: #6628

## Description of Changes
- Adds the community alerts and subscriptions... 
- 

## Test Plan
 - You have to enable the knock in-app notifications feature toggles then go to the notification settings page and we have a new layout.
  - 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/notification_settings2/SubEntry.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/notification_settings2/SubEntry.tsx:40]: okay so for couple last months we were moving from old approach of fetching data to react-query. We decided that all our "interfaces" will sit in `state/api/*` directories. I would love to stick to this regardless we use trpc, reqct-query or whatever. The reason for this is that we don't do the mess in the components and we keep all the backend points of contact in one structured place. 

Please take a look on `state/api/contests` [as an example](https://github.com/hicommonwealth/commonwealth/blob/6ff154894b84fedaabff9bbb78a247fda147cc93/packages/commonwealth/client/scripts/state/api/contests/getContests.ts#L8). It uses trpc but we still create files and hooks in the way we did in the past.  

So please create state/api/subscriptions directory and inside
- deleteThreadSubscription.ts
- getThreadSubscriptions.ts
and inside create a hook that wraps trpc. Please follow the pattern

File: packages/commonwealth/client/scripts/views/pages/notification_settings2/SubEntry.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/notification_settings2/SubEntry.tsx:42]: I find those useCallbacks redundant. What's is the reason for adding them?

File: packages/commonwealth/client/scripts/views/pages/notification_settings2/index.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/notification_settings2/index.tsx:13]: take a look on my another comment regarding creating hooks and proper directory structure

File: packages/commonwealth/client/scripts/views/pages/notification_settings2/index.scss
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/notification_settings2/index.scss:18]: we should have main class at the top and then nest other classes inside 

```
.NotificationSettingsPage2 {
  .SubEntry{}
  .SubHeader{}

  etc
}
```

File: libs/model/src/models/thread_subscriptions.ts
Comment (mzparacha) [inline libs/model/src/models/thread_subscriptions.ts:16]: do we need this?

File: packages/commonwealth/server/routes/status.ts
Comment (mzparacha) [inline packages/commonwealth/server/routes/status.ts:384]: `KNOCK_SIGNING_KEY` is still marked as optional in `/libs/adapters/src/config.ts`, not sure If we should remove this warning?

File: packages/commonwealth/client/scripts/views/pages/NotificationSettingsOld/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettingsOld/index.tsx:170]: The lint disables in this file are easily fixable, let's try not to disable link rules where it's unnecessary.

File: packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx:51]: features flags like `knockInAppNotifications`, should come from Router.tsx per existing convention. ex:

https://github.com/hicommonwealth/commonwealth/blob/9885315df5941a3a57c82abccb911f4471ad587d/packages/commonwealth/client/scripts/navigation/Router.tsx#L31-L35

File: packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx:40]: same as https://github.com/hicommonwealth/commonwealth/pull/7910/files#r1639881925

File: libs/schemas/src/entities/notification.schemas.ts
Comment (mzparacha) [inline libs/schemas/src/entities/notification.schemas.ts:99]: `.merge` seems unnecessary, will this work?

```
z.object({
  user_id: PG_INT,
  community_id: z.string(),
  created_at: zDate.optional(),
  updated_at: zDate.optional(),
  Community: Community.pick({
    id: true,
    name: true,
    icon_url: true,
  }).optional(),
});
```

File: libs/schemas/src/entities/notification.schemas.ts
Comment (mzparacha) [inline libs/schemas/src/entities/notification.schemas.ts:72]: same as above, will this work?
```
Thread: Thread.pick({
    id: true,
    community_id: true,
    address_id: true,
    title: true,
    comment_count: true,
    created_at: true,
    url: true,
  Community: Community.pick({
    id: true,
    name: true,
    icon_url: true,
  }),
  Address: Address.pick({
    id: true,
    profile_id: true,
    user_id: true,
    address: true,
  }),
  }).optional(),
```

File: libs/schemas/src/entities/notification.schemas.ts
Comment (mzparacha) [inline libs/schemas/src/entities/notification.schemas.ts:47]: do we need to coerce here? all other `lib/schemas/src/entities` seem to just use `z.date()...`

File: libs/schemas/src/utils.ts
Comment (mzparacha) [inline libs/schemas/src/utils.ts:44]: There are a lot of instances where this is used. 

The newer `z.coerce.date();` doesn't give errors on `numbers` and `null` types -- but the removed validation does. Not sure what the effects would be, I would suggest we find out the effects or revert this.

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx:25]: I think we should use the newer notification settings in this component, or use different notification settings config depending on `FLAG_KNOCK_INTEGRATION_ENABLED`.

Else when `FLAG_KNOCK_INTEGRATION_ENABLED` is enabled in prod, this component would give undesired results

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/useCommunityAlerts.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/useCommunityAlerts.ts:1]: i think this should be inside `/state/api`

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/useThreadSubscriptions.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/useThreadSubscriptions.ts:1]: same as https://github.com/hicommonwealth/commonwealth/pull/7910/files#r1639926904

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:128]: this seems unnecessary, as `app.config.chains` gets defined, in the first few API calls after page load. 

Is there a case where this logic gets triggered?

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:57]: lets remove console

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:72]: we normally show `<PageNotFound/>` when user visits auth gated pages without any authentication.

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:42]: nice

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:32]: lets put trpc queries into `state/api`

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/CommunityEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/CommunityEntry.tsx:25]: lets put these in `state/api` per existing convention

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/CommunityEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/CommunityEntry.tsx:64]: lets use css classes, instead of inline styles

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx:42]: lets put it in `state/api` per existing convention

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx:24]: whats the purpose of `!` after values?

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx:67]:
```diff
thread?.Community?.icon_url
thread?.Community?.name
```

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx:74]:
```diff
- thread.Community.name
+ thread?.Community?.name
```

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx:89]: we are showing thread time in `DD/MM/YYYY h:mm A` per https://github.com/hicommonwealth/commonwealth/issues/8036#issuecomment-2147517393

**Review by mzparacha — COMMENTED** [review]
Left some feedback above, wasn't able to test as it required some knock config which I am not sure how to do.

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/CommunityEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/CommunityEntry.tsx:62]: this should be replaced by `CommunityInfo.tsx` as it would also handle community linking

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx:1]: this doesn't match designs per https://github.com/hicommonwealth/commonwealth/issues/6628

currently it looks like this

image

we can use the `<ThreadCard/>` component for thread rendering here.

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:110]: Possible enhancement: we mostly add the active tab to URL, ex: `?tab=activeTabName` and use it to set the active tab. This also helps in page bookmarks etc...

File: libs/schemas/src/entities/notification.schemas.ts
Comment (mzparacha) [inline libs/schemas/src/entities/notification.schemas.ts:85]: lets remove this if not needed

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:29]: lets remove unused vars

File: packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/navigation/CommonDomainRoutes.tsx]: please also update this in `CustomDomainRoutes.tsx`

File: packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx:40]: please update this similar to how you did with `CommonDomainRoutes`

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:71]: please use `useUserLoggedIn` hook to get app loggedin state

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:111]: sounds good. I wouldn't saw its a blocker but would be great to get this in. If not lets create a ticket to address this later.

File: packages/commonwealth/server/routes/status.ts
Comment (mzparacha) [reply packages/commonwealth/server/routes/status.ts:385]: @burtonator

File: packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx:40]: @burtonator

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx:25]: > You mean just another flag? I'm trying to avoid creating too many. I think we're at 5-6 so far.

Actually no, I meant that the `PersonalInformationForm` uses the old way to subscribe user to notifications, once knock is enabled in app we would want to use knock notification (ex: `trpc.subscription.createCommunityAlert` and others) instead of the old notifications in this component. Otherwise this component will break/give-undesired-results when knock is enabled

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:111]: OK created a ticket for it: 

https://github.com/hicommonwealth/commonwealth/issues/8191

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/NotificationSettings/SubscriptionEntry.tsx:1]: hm, i still see a lot of differences with other designs like our existing thread card which looks like this (notice the hover/link and other behaviours)

https://github.com/hicommonwealth/commonwealth/assets/51641047/fd10e03d-d122-40a9-a24c-891f9cf32233

vs 

https://github.com/hicommonwealth/commonwealth/assets/51641047/d2bdc03f-0241-4fa2-951d-fdd7275c55d6

.

Anyways if the design team still approved the pr implemented design then all good from my side. Please feel free to resolve

**Review by Israellund — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx:25]: This component won't be rendered and we will use the new notifications page which uses tRPC. The old page can be removed at that point. 

I think we're good here.

File: packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx
Comment (burtonator) [reply packages/commonwealth/client/scripts/navigation/CustomDomainRoutes.tsx]: Fixed this but not sure why we have TWO routes here...

File: packages/commonwealth/server/routes/status.ts
Comment (burtonator) [reply packages/commonwealth/server/routes/status.ts:385]: Yes.  This is removed.  Are you looking at a stale diff?

File: libs/model/src/models/thread_subscriptions.ts
Comment (burtonator) [reply libs/model/src/models/thread_subscriptions.ts:16]: Yes... I guess.  It's part of the convention on the backend. We're using these in multiple places and I think the point is that we can add these in the future.

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx:25]: Actually `PersonalInformationForm` is rendered when a user creates a new account while `FLAG_USER_ONBOARDING_ENABLED` is `true`.

The said I think this will cause a bug is because `PersonalInformationForm` uses the `useNotificationSettings` from old notifications page which enables/disables notifications using old older `app.user.notifications.` methods

File: packages/commonwealth/server/routes/status.ts
Comment (mzparacha) [reply packages/commonwealth/server/routes/status.ts:385]: I mean shouldn't we keep it as `KNOCK_SIGNING_KEY`  is optional

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx:25]: Where do you see PersonalInformationForm?  I don't have that as a symbol in my entire repo except for personalInformationFormValidation

Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx:25]: Ah mb, sry for the confusion, the component name is `PersonalInformationStep.tsx`

cc: @burtonator

File: packages/commonwealth/server/routes/status.ts
Comment (burtonator) [reply packages/commonwealth/server/routes/status.ts:385]: It's required if FLAG_KNOCK_INTEGRATION_ENABLED is on but I think I removed it in the first place because either you or Marcin felt it was too aggressive :) I think it was Marcin though.

File: packages/commonwealth/client/scripts/state/api/trpc/subscription/useCommunityAlertsQuery.ts
Comment (masvelio) [inline packages/commonwealth/client/scripts/state/api/trpc/subscription/useCommunityAlertsQuery.ts:4]: we should pass staleTime param to the query, otherwise, it will be called way to often. Find `staleTime:` in the code and follow the pattern

File: packages/commonwealth/client/scripts/state/api/trpc/subscription/useThreadSubscriptionsQuery.ts
Comment (masvelio) [inline packages/commonwealth/client/scripts/state/api/trpc/subscription/useThreadSubscriptionsQuery.ts:4]: same here for a stale time

File: packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx:25]: can we get another look on this, please @burtonator

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:8]: we should create index.ts file in api/trpc/subscription that would import and export all the hooks, then, you could simply do 

`import { useRegisterClientRegistrationTokenMutation } from 'state/api/trpc/subscription/';`

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:1]: I think I mentioned it already in previous PR but the file should not be called `index.tsx`. This is the convention we use and this really helps with searching files etc. index file should be used for exposing files from the directory so it is simpler to import them later.
This file should be `NotificationSettings.tsx`

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.scss
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.scss:1]: same here, this file should be called `NotificationSettings.scss`

**Review by masvelio — APPROVED** [review]
added few small comments, approving in advance, but because @mzparacha did great review, I want him also to give it ✅  before it will be merged

Thanks for hard job and all those changes

**Review by mzparacha — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:8]: This kind of defeats the point because the idea was that we would wrap them with logic and I think throwing them all into the same file is probably not the best idea.

File: packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx:1]: Going to fix this in another PR: https://github.com/hicommonwealth/commonwealth/issues/8212

**Pull request merged**
